rendermodules/em_montage_qc/detect_montage_defects.py

```diff
@@ -0,0 +1,266 @@
+    [pre_ridx.insert(i, (ts.minX, ts.minY, ts.maxX, ts.maxY)) for i, ts in enumerate(pre_tilespecs)]
+
+    pre_tileIds = {}
+    for i, ts in enumerate(pre_tilespecs):
+        pre_tileIds[ts.tileId] = i
+        nodes = list(pre_ridx.intersection((ts.minX, ts.minY, ts.maxX, ts.maxY)))
+        nodes.remove(i)
+        [G1.add_edge(i, node) for node in nodes]
+
+
+    # G1 contains the prestitched_stack tiles and the degree of each node representing
+    # the number of tiles that overlap
+    # This overlap count has to match in the poststitched_stack
+
+    G2 = nx.Graph()
+    post_ridx = rindex.Index()
+
+    [post_ridx.insert(pre_tileIds[ts.tileId], (ts.minX, ts.minY, ts.maxX, ts.maxY)) for ts in tilespecs]
+
+    for ts in tilespecs:
+        i = pre_tileIds[ts.tileId]
+        nodes = list(post_ridx.intersection((ts.minX, ts.minY, ts.maxX, ts.maxY)))
+        nodes.remove(i)
+        [G2.add_edge(i, node) for node in nodes]
+
+
+    # Now G1 and G2 have the same index for the same tileId
+    # comparing the degree of each node pre and post stitching should reveal stitching gaps
+    #tileIds = nx.get_node_attributes(G2, 'tileId')
+    gap_tiles = []
+    for n in G2.nodes():
+        if G1.degree(n) > G2.degree(n):
+            tileId = pre_tileIds.keys()[pre_tileIds.values().index(n)]
+            gap_tiles.append(tileId)
+
+    return gap_tiles
+
+def detect_stitching_mistakes(render, prestitched_stack, poststitched_stack, match_collection, match_collection_owner, residual_threshold, neighbor_distance, min_cluster_size, zvalues, pool_size=20):
+    mypartial1 = partial(detect_disconnected_tiles, 
+                         render, 
+                         prestitched_stack, 
+                         poststitched_stack)
+    mypartial2 = partial(detect_stitching_gaps, 
+                         render, 
+                         prestitched_stack, 
+                         poststitched_stack)
+    mypartial3 = partial(detect_seams, 
+                         render, 
+                         poststitched_stack, 
+                         match_collection, 
+                         match_collection_owner, 
+                         residual_threshold=residual_threshold, 
+                         distance=neighbor_distance,
+                         min_cluster_size=min_cluster_size)
+
+    disconnected_tiles = []
+    gap_tiles = []
+    seam_centroids = []
+
+    with renderapi.client.WithPool(pool_size) as pool:
+        disconnected_tiles = pool.map(mypartial1, zvalues)
+        gap_tiles = pool.map(mypartial2, zvalues)
+        seam_centroids = pool.map(mypartial3, zvalues)
+
+    #for z in zvalues:
+    #    seam_centroids = detect_seams(render, poststitched_stack, match_collection, match_collection_owner, z, distance=80)
+
+    return disconnected_tiles, gap_tiles, seam_centroids
+
+
+class DetectMontageDefectsModule(RenderModule):
+    def __init__(self, schema_type=None, *args, **kwargs):
```

Comment:
This can be simplified as 
```
class DetectMontageDefectsModule(RenderModule):
   default_schema = DetectMontageDefectsParameters
   default_output_schema = DetectMontageDefectsParametersOutput